src/test/java/com/team1/dealerApp/video/movie/MovieServiceTest.java

```diff
@@ -0,0 +1,223 @@
+package com.team1.dealerApp.video.movie;
+
+
+import com.team1.dealerApp.video.Genre;
+import com.team1.dealerApp.video.VideoStatus;
+import org.apache.coyote.BadRequestException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Year;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class MovieServiceTest {
+
+    // Simula il MovieRepository
+    @Mock
+    private MovieRepository movieRepository;
+
+    // Simula il MovieMapper
+    @Mock
+    private MovieMapper movieMapper;
+
+    // Simula il MovieUpdater
+    @Mock
+    private MovieUpdater<Object> movieUpdater;
+
+    // Inietta il MovieService da testare
+    @InjectMocks
+    private MovieService movieService;
+
+    MovieDTO movieDTO;
+    MovieDTO movieDTO2;
+    Movie movie;
+    Movie movie2;
+
+    @BeforeEach
+    public void setup() {
+        // Inizializza i mock prima di ogni test
+        MockitoAnnotations.openMocks(this);
+        List<String> castMovie = Arrays.asList("Elijah Wood", "Ian McKellan", "Orlando Bloom");
+        List<String> castMovie2 = Arrays.asList("Ed Wynn", "Richard Haydn", "Kathryn Beaumont");
+
+        movieDTO = new MovieDTO("Il signore degli anelli: il ritorno del Re", 110, Genre.FANTASY, castMovie, "Peter Jackson", Year.of(2003), 30.00, 10.00, "Il ritorno del Re", 4.0f, VideoStatus.PURCHASABLE);
```

Comment:
in questo caso, movieDTO potrebbe chiamarsi "purchasableMovieDTO" e movieDTO2 "rentableMovieDTO", poichè la differenza credo sia solo quella. l'importante è capire quando un oggetto va utilizzato nei test.